interfaces/apparmor/template_vars.go

```diff
@@ -72,38 +69,60 @@ func legacyVariables(appInfo *snap.AppInfo) string {
 // @{SNAP_NAME}=app.SnapName
 //
 // ...have everything work correctly?
-func modernVariables(appInfo *snap.AppInfo) string {
-	var buf bytes.Buffer
-	fmt.Fprintf(&buf, "@{APP_NAME}=\"%s\"\n", appInfo.Name)
-	fmt.Fprintf(&buf, "@{APP_SECURITY_TAG}=\"%s\"\n", interfaces.SecurityTag(appInfo))
-	fmt.Fprintf(&buf, "@{SNAP_NAME}=\"%s\"\n", appInfo.Snap.Name)
-	fmt.Fprintf(&buf, "@{INSTALL_DIR}=\"{/snaps,/gadget}\"\n")
-	return buf.String()
+func writeModernVariables(buf *bytes.Buffer, appInfo *snap.AppInfo) {
+	fmt.Fprintf(buf, "@{APP_NAME}=\"%s\"\n", appInfo.Name)
+	fmt.Fprintf(buf, "@{APP_SECURITY_TAG}=\"%s\"\n", interfaces.SecurityTag(appInfo))
+	fmt.Fprintf(buf, "@{SNAP_NAME}=\"%s\"\n", appInfo.Snap.Name)
+	fmt.Fprintf(buf, "@{INSTALL_DIR}=\"{/snaps,/gadget}\"\n")
 }
 
-// aaHeader returns the topmost part of the generated apparmor profile.
+// writeProfileAttach writes the profile attachment line of the apparmor profile.
 //
-// The header contains a few lines of apparmor variables that are referenced by
-// the template as well as the syntax that begins the content of the actual
-// profile. That same content also decides if the profile is enforcing or
-// advisory (complain). This is used to implement developer mode.
-func (b *Backend) aaHeader(appInfo *snap.AppInfo, developerMode bool) []byte {
-	var text string
-	// Use a different template when requested.
-	if len(b.CustomTemplate) == 0 {
-		text = strings.TrimRight(defaultTemplate, "\n}")
-		// TODO: switch to modernVariables() after the default template is
-		// compatible with them.
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
-	} else {
-		text = strings.TrimRight(b.CustomTemplate, "\n}")
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
-	}
+// The line typically looks like this:
+//
+//   profile "snap.http.GET" (attach_disconnected) {
+func writeProfileAttach(buf *bytes.Buffer, appInfo *snap.AppInfo, developerMode bool) {
+	fmt.Fprintf(buf, "profile \"%s\" (attach_disconnected", interfaces.SecurityTag(appInfo))
 	if developerMode {
 		// XXX: This needs to be verified by security team.
-		text = strings.Replace(text, "(attach_disconnected)", "(attach_disconnected,complain)", 1)
+		fmt.Fprintf(buf, ",complain")
+	}
+	fmt.Fprintf(buf, ") {\n")
+}
+
+const (
```

Comment:
Using var and []byte(...) here will make the logic below slightly simpler and nicer to read.